src/components/DivingCylinderSet/newDivingCylinderSet.tsx

```diff
@@ -1,8 +1,166 @@
+import { Field, FieldArray, FieldArrayRenderProps, Formik, Form } from 'formik';
+import React, { useCallback } from 'react';
+import { Button, Col, InputGroup } from 'react-bootstrap';
+import { BsTrash } from 'react-icons/bs';
+
+import '../../styles/divingCylinderSet/newDivingCylinderSet.css';
+import { IconButton } from '../common/IconButton';
+
+type DivingCylinder = {
+  volume: string;
+  material: string;
+  maximumFillingPressure: string;
+  serialNumber: string;
+  inspectionYear: string;
+};
+
+type DivingCylinderSet = {
+  divingCylinderSetName: string;
+  divingCylinders: DivingCylinder[];
+};
+
+const EmptyDivingCylinder: DivingCylinder = {
+  volume: '',
+  material: '',
+  maximumFillingPressure: '',
+  serialNumber: '',
+  inspectionYear: '',
+};
+
+const NewDivingCylinderRow = (
+  { remove, push }: FieldArrayRenderProps,
+  index: number,
+  lastItem: boolean
+): JSX.Element => {
+  return (
+    <div key={index}>
+      <div className="gridRow">
+        <div className="labelAndInput">
+          <InputGroup as={Col}>
+            <Field
+              className="form-control"
+              name={`divingCylinders.${index}.volume`}
+            />
+            <InputGroup.Text>l</InputGroup.Text>
+          </InputGroup>
+        </div>
+        <div className="labelAndInput">
+          <InputGroup as={Col}>
+            <Field
+              type="text"
+              list={`divingCylinders.${index}.material`}
+              className="form-select"
+              name={`divingCylinders.${index}.material`}
+            />
+            <datalist id={`divingCylinders.${index}.material`}>
+              <option>Alumiini</option>
+              <option>Hiilikuitu</option>
+              <option>Teräs</option>
+            </datalist>
+          </InputGroup>
+        </div>
+        <div className="labelAndInput">
+          <InputGroup as={Col}>
+            <Field
+              type="number"
+              list={`divingCylinders.${index}.maximumFillingPressures`}
+              className="form-select"
+              name={`divingCylinders.${index}.maximumFillingPressure`}
+            />
+            <datalist id={`divingCylinders.${index}.maximumFillingPressures`}>
```

Comment:
Tääkin vois varmaan olla puhdas pudotusvalikko, tahi sitten numerokenttä. Nyt tämä arvo tarvitsee validoida erikseen.

Reply:
Numerokenttä it is then. Huom. Firefox antaa silti kirjoittaa kenttään muutakin kuin numeroita, chrome ei =)